### PR TITLE
[Ready for review] recent revisions only fetch on render #244

### LINKED ISSUE #244: recent revisions only fetch on render
When a user types in a search term, then deletes it, the recent revisions are no longer available, because the `searchResults` state has been changed.

Expected behavior: recent revisions still available after entering search term and deleting it.

### PR DESCRIPTION
**Pull request by NikolaDojic**
closes #244

### CODE REVIEW

**Comment by esanuandra** [comment]
Hey @NikolaDojic! Thank you for your contribution. When you finish this add the `ready for review` label.

**Comment by NikolaDojic** [comment]
Hey, not a problem :)

Screenshot_20221207_123141

@esanuandra, I don't think that I can, unless I'm missing something.  I think writing permission is needed for that.

**Comment by NikolaDojic** [comment]
ready for review

File: src/__tests__/utils/test-utils.tsx [hunk 1]
```diff
@@ -7,9 +7,17 @@ import { Router } from 'react-router-dom';
 import type { Store } from '../../common/store';
 import SnackbarCloseButton from '../../components/Shared/SnackbarCloseButton';
 
+type ChildrenProps = { children: React.ReactElement };
+
+const createStoreProvider = (store: Store) => {
+  return function SroreProvider({ children }: ChildrenProps) {
```
Comment (esanuandra) [inline src/__tests__/utils/test-utils.tsx:13]: typo?

File: src/__tests__/hooks/useHandleChangeSearch.test.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    });
+    store.dispatch(setInputError(testError));
+    const { search: searchSlice } = store.getState();
+    expect(searchSlice.inputHelperText).toBe('test error');
+    result.current.handleChangeSearch(createEvent(testInput));
+    const { search: updatedSearchSlice } = store.getState();
+    expect(updatedSearchSlice.inputHelperText).toBe('');
+  });
+
+  it('shold update inputHelperText to contain an error if search text does not match email or hash', () => {
```
Comment (esanuandra) [inline src/__tests__/hooks/useHandleChangeSearch.test.ts:66]: typo

**Review by esanuandra — CHANGES_REQUESTED** [review]
Looks good. Could you please look into the failing for codecov/project? Maybe a rebase would help.

File: src/__tests__/utils/test-utils.tsx (hunk 1 above)
Comment (NikolaDojic) [reply src/__tests__/utils/test-utils.tsx:13]: thanks, and sorry about that

**Comment by NikolaDojic** [comment]
> Looks good. Could you please look into the failing for codecov/project? Maybe a rebase would help.

@esanuandra the codecov report kind of baffles me. 
Screenshot_20221212_112133

it states that all lines are covered in the affected files, but still fails. can we trigger a re-run?

**Review by esanuandra — APPROVED** [review]
nice work

**Review by kimberlythegeek — APPROVED** [review]

**Pull request merged**